server/src/guest/guest.controller.ts

```diff
@@ -190,4 +195,56 @@ export class GuestController {
 	async remove(@Param('uuid') uuid: string): Promise<DeleteResult> {
 		return this.guestService.delete(uuid);
 	}
+
+	/**
+	 * If no admin exists, make the user with the provided google Id or create one and make them an admin
+	 * @param googleAuthJWT A JWT generated by google with relevant user data
+	 * @returns The new admin user's primary data
+	 */
+	@Post('createadmin')
+	async createAdmin(
+		@Body() googleAuthJWT: { token: string },
+	): Promise<primaryData> {
+		// Check if there are already any admins
+		if (await this.adminService.doAnyAdminsExist()) {
+			throw new HttpException(
+				{
+					status: HttpStatus.FORBIDDEN,
+					error: 'There are already admins',
+				},
+				HttpStatus.FORBIDDEN,
+			);
+		}
+
+		// Get user info from google and check if a user already has the google id provided
+		const userIdentity: RecursivePartial<guestIdentity> =
+			await this.authService.getIdentityFromGoogleToken(
+				googleAuthJWT.token,
+			);
+		if (!userIdentity) throw new UnauthorizedException();
+
+		let user = await this.contactService.getUserByGoogleAuthID(
+			userIdentity.googleAuthId,
+		);
+
+		// Create a new user if one with the google ID does not exist
+		if (!user) {
+			user = await this.guestService.create({
+				firstName: userIdentity.firstName || 'Admin',
+				lastName: userIdentity.lastName || 'User',
+				uuid: null,
+			});
+
+			if (!user) throw new InternalServerErrorException();
```

Comment:
What situations might result in a user not existing at this point? Like, what can make the guest service creation fail?

Reply:
Or is this just a cya thing

Reply:
Random database problems